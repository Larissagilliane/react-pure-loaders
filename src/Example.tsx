import * as React from 'react';

export const Example = (props: any) => {
  return <span {...props}>Hello World!</span>;
};