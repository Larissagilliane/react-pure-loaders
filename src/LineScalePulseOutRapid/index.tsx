import { css } from 'glamor';
import * as React from 'react';
import style from './style';

export interface IGeneralPropTypes {
  loading: boolean;
}

export class LineScalePulseOutRapid extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { loading } = this.props;

    if (loading) {
      return (
        <div {...css(style)}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
