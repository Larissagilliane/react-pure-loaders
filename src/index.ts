/**
 * Dots
 */
export { BallBeat } from './BallBeat';
export { BallClipRotate } from './BallClipRotate';
export { BallClipRotateMultiple } from './BallClipRotateMultiple';
export { BallClipRotatePulse } from './BallClipRotatePulse';
export { BallGridBeat } from './BallGridBeat';
export { BallGridPulse } from './BallGridPulse';
export { BallPulse } from './BallPulse';
export { BallPulseRise } from './BallPulseRise';
export { BallPulseRound } from './BallPulseRound';
export { BallPulseSync } from './BallPulseSync';
export { BallRotate } from './BallRotate';
export { BallScale } from './BallScale';
export { BallScaleMultiple } from './BallScaleMultiple';
export { BallScaleRandom } from './BallScaleRandom';
export { BallScaleRipple } from './BallScaleRipple';
export { BallScaleRippleMultiple } from './BallScaleRippleMultiple';
export { BallSpinFadeLoader } from './BallSpinFadeLoader';
export { BallSpinLoader } from './BallSpinLoader';
export { BallTrianglePath } from './BallTrianglePath';
export { BallZigZag } from './BallZigZag';
export { BallZigZagDeflect } from './BallZigZagDeflect';

/**
 * Lines
 */
export { LineScale } from './LineScale';
export { LineScaleParty } from './LineScaleParty';
export { LineScalePulseOut } from './LineScalePulseOut';
export { LineScalePulseOutRapid } from './LineScalePulseOutRapid';
export { LineScaleRandom } from './LineScaleRandom';
export { LineSpinFadeLoader } from './LineSpinFadeLoader';

/**
 * Misc
 */
export { CubeTransition } from './CubeTransition';
export { Pacman } from './Pacman';
export { SemiCircleSpin } from './SemiCircleSpin';
export { SquareSpin } from './SquareSpin';
export { TriangleSkewSpin } from './TriangleSkewSpin';
