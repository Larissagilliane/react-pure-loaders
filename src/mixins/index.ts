import { css } from 'glamor';
import {
  BALL_SIZE,
  LINE_HEIGHT,
  LINE_WIDTH,
  MARGIN,
  PRIMARY_COLOR,
} from '../variables';

interface IGrid {
  [id: string]: object;
}

export function globalBg() {
  return {
    backgroundColor: PRIMARY_COLOR,
  };
}

export function globalAnimation() {
  return {
    animationFillMode: 'both',
  };
}

export function balls() {
  return css(
    {
      borderRadius: '100%',
      height: BALL_SIZE,
      margin: MARGIN,
      width: BALL_SIZE,
    },
    globalBg(),
  );
}

export function ballGrid() {
  let i = 1;
  const grid: IGrid = {};
  for (; i < 10; i += 1) {
    const delay = ((parseInt((Math.random() * 100).toString(), 10) / 100) - 0.2);
    const duration = ((parseInt((Math.random() * 100).toString(), 10) / 100) + 0.6);

    grid[` > div:nth-child(${i})`] = {
      animationDelay: `${delay}s`,
      animationDuration: `${duration}s`,
    };
  }

  return grid;
}

export function lines() {
  return css(
    {
      borderRadius: '2px',
      height: LINE_HEIGHT,
      margin: MARGIN,
      width: LINE_WIDTH,
    },
    globalBg(),
  );
}
