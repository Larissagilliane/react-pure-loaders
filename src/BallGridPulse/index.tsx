import { css } from 'glamor';
import * as React from 'react';
import { IGeneralPropTypes } from '../mixins';
import style from './style';

export class BallGridPulse extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { color, loading } = this.props;

    if (loading) {
      return (
        <div {...css(style(color))}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
