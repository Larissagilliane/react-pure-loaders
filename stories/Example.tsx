import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs';

import { Example } from '../src/Example'

const stories = storiesOf('Example', module);

const divWrapperStyle: object = {
  fontFamily: '"Helvetica Neue", Helvetica, Roboto, Arial, sans-serif',
  fontWeight: 'normal',
  lineHeight: '1.5',
  color: '#0a0a0a',
  MozOsxFontSmoothing: 'grayscale',
};

stories.addDecorator(withKnobs);

stories.add('Basics', () => {
  return (
    <div style={divWrapperStyle}>
      <Example/>
    </div>
  );
});
