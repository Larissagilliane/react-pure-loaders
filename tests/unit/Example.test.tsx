import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Example } from '../../src/Example';

describe('<Example>', () => {
  test('Example component should match snapshot', () => {
    const wrapper = shallow(<Example/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});