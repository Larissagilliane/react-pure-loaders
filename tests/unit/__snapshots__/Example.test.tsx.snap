// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Example> Example component should match snapshot 1`] = `
<span>
  Hello World!
</span>
`;
